hub/core/dataset/dataset.py

```diff
@@ -403,7 +403,8 @@ def commit(self, message: Optional[str] = None) -> None:
             str: the commit id of the stored commit that can be used to access the snapshot.
         """
         commit_id = self.version_state["commit_id"]
-        try_flushing(self)
+        initial_autoflush = self.storage.autoflush
```

Comment:
always disable autoflush during commit and enable it back at the end if required